service/history/ndc/workflow_state_replicator.go

```diff
@@ -336,11 +336,14 @@ BackfillLoop:
 
 		if historyBlob.nodeID <= lastBatchNodeID {
 			// The history batch already in DB.
-			currentAncestor := sortedAncestors[sortedAncestorsIdx]
-			if historyBlob.nodeID >= currentAncestor.GetEndNodeId() {
-				// update ancestor
-				ancestors = append(ancestors, currentAncestor)
-				sortedAncestorsIdx++
+			if len(sortedAncestors) > 0 {
+				currentAncestor := sortedAncestors[sortedAncestorsIdx]
```

Comment:
Can we guarantee that sortedAncestorsIdx < len(sortedAncestors)?

Reply:
sure. good point.